llvm/lib/Target/WebAssembly/AsmParser/WebAssemblyAsmTypeCheck.cpp

```diff
@@ -70,14 +70,9 @@ void WebAssemblyAsmTypeCheck::dumpTypeStack(Twine Msg) {
 }
 
 bool WebAssemblyAsmTypeCheck::typeError(SMLoc ErrorLoc, const Twine &Msg) {
-  // Once you get one type error in a function, it will likely trigger more
-  // which are mostly not helpful.
-  if (TypeErrorThisFunction)
```

Comment:
This change looks reasonable to me, but I do wonder if we should maybe do as thomas suggested in the meeting and only report one error per block?   So this would be `TypeErrorThisBlock`?

I don't feel to strongly though.

Reply:
We can probably make it a follow-up or an improvement, but I feel that's more confusing, given that no other tools work that way. As I summarized in https://github.com/llvm/llvm-project/pull/109705#issuecomment-2369243689,
- Binaryen parser: Bail out at first error
- Binaryen validator: Report all errors
- Wabt parser: Report all errors
So no other tools do "report future errors but only one error per block" thing.

Reply:
I guess Binaryen's validator can produce the same kind of false positive error messages if the parser hooks up the parents and children incorrectly due to the buggy input.

Reporting all the errors seems fine, but reporting just one error per block might be a nice way to generate fewer false positives. The other tools could benefit from that, too, so this idea isn't specific to LLVM in any way. OTOH, the number of users of this functionality is tiny, so it's probably not worth investing much in the quality of the error messages.

Reply:
Yeah I agree that can be useful, but I also agree that it's not clear how much effort is worth spending on this. Will follow up if that functionality seems necessary later.